### PR TITLE
Allow specifying a collection of colors to scatterplots

### LINKED ISSUE #1986: `scanpy.pl.scatter` fails if list is provided for `color`
- [x] I have checked that this issue has not already been reported.
- [x] I have confirmed this bug exists on the latest version of scanpy.
- [ ] (optional) I have confirmed this bug exists on the master branch of scanpy.

### Minimal code sample

```python
import numpy as np
import scanpy as sc

adata = sc.datasets.paul15()
adata.obs['batch'] = np.random.randint(3, size=adata.n_obs)
adata.obs['X'] = adata.X[:,1]
adata.obs['Y'] = adata.X[:,2]

sc.pl.scatter(adata, x='X', y='Y', color='batch')  # works
sc.pl.scatter(adata, x='X', y='Y', color=['paul15_clusters', 'batch']) # fails
```

```pytb
Traceback (most recent call last):
  File "/home/mumichae/miniconda3/envs/opsca/lib/python3.8/site-packages/IPython/core/interactiveshell.py", line 3441, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-52-9844c466c985>", line 1, in <module>
    sc.pl.scatter(adata, x='X', y='Y', color=['paul15_clusters', 'batch'])
  File "/home/mumichae/miniconda3/envs/opsca/lib/python3.8/site-packages/scanpy/plotting/_anndata.py", line 131, in scatter
    and (color is None or color in adata.obs.keys() or color in adata.var.index)
  File "/home/mumichae/miniconda3/envs/opsca/lib/python3.8/site-packages/pandas/core/indexes/base.py", line 4572, in __contains__
    hash(key)
TypeError: unhashable type: 'list'
```

#### Versions

<details>

```
-----
anndata     0.7.6
scanpy      1.8.1
sinfo       0.3.1
-----
IPython             7.25.0
PIL                 8.3.1
anndata             0.7.6
backcall            0.2.0
backend_interagg    NA
beta_ufunc          NA
binom_ufunc         NA
cffi                1.14.6
colorama            0.4.4
console_thrift      NA
cycler              0.10.0
cython_runtime      NA
datalore            NA
dateutil            2.8.1
decorator           4.4.2
defusedxml          0.7.1
h5py                2.10.0
igraph              0.9.6
ipython_genutils    0.2.0
jedi                0.18.0
joblib              1.0.1
kiwisolver          1.3.1
llvmlite            0.36.0
louvain             0.7.0
matplotlib          3.4.2
mpl_toolkits        NA
natsort             7.1.1
nbinom_ufunc        NA
numba               0.53.1
numexpr             2.7.3
numpy               1.18.1
packaging           20.9
pandas              1.3.1
parso               0.8.2
pexpect             4.8.0
pickleshare         0.7.5
pkg_resources       NA
prompt_toolkit      3.0.19
psutil              5.8.0
ptyprocess          0.7.0
pycparser           2.20
pydev_console       NA
pydev_ipython       NA
pydevconsole        NA
pydevd_file_utils   NA
pydevd_plugins      NA
pydevd_tracing      NA
pygments            2.9.0
pynndescent         0.5.4
pyparsing           2.4.7
pytz                2021.1
scanpy              1.8.1
scipy               1.7.0
sinfo               0.3.1
sitecustomize       NA
six                 1.16.0
sklearn             0.24.2
sphinxcontrib       NA
tables              3.6.1
texttable           1.6.4
tqdm                4.61.2
traitlets           5.0.5
umap                0.5.1
wcwidth             0.2.5
-----
Python 3.8.10 | packaged by conda-forge | (default, May 11 2021, 07:01:05) [GCC 9.3.0]
Linux-5.13.10-200.fc34.x86_64-x86_64-with-glibc2.10
16 logical CPU cores
-----
Session information updated at 2021-08-26 10:01

```

</details>

### PR DESCRIPTION
**Pull request by flying-sheep**
- [x] Closes #1986
- [x] Tests included or not required because:

- [ ] Release notes not necessary because:

This PR moves conversion from `str` to `list[str]` up and changes the check if `color` is in `obs.columns`/`var_names` to a check that can handle collections instead of just a single `color` value.

### CODE REVIEW

File: src/scanpy/plotting/_anndata.py [hunk 1]
```diff
@@ -148,69 +149,101 @@ def scatter(
...
-        and (color is None or color in adata.var.columns or color in adata.obs.index)
-    ):
-        adata_T = adata.T
-        axs = _scatter_obs(
-            adata=adata_T,
-            **{name: val for name, val in args.items() if name != "adata"},
-        )
+    if _check_if_annotations(adata, "var", x=x, y=y, colors=color, use_raw=use_raw):
+        args_t = args.copy()
+        args_t["adata"] = adata.T
```
Comment (ilan-gold) [inline src/scanpy/plotting/_anndata.py]: Out of curiosity, why not
```suggestion
        args_t = { **args, "adata": adata.T }
```

**Review by ilan-gold — COMMENTED** [review]
Out of curiosity, why no test for a longer collection of colors?

File: src/scanpy/plotting/_anndata.py (hunk 1 above)
Comment (flying-sheep) [reply src/scanpy/plotting/_anndata.py]: huh, good point, I thought that’d cause something like the “parameter specified multiple times” error, but it doesn’t.

**Comment by flying-sheep** [comment]
> why no test for a longer collection of colors?

what do you mean? just a longer list? what would that test?

**Comment by flying-sheep** [comment]
pre-commit.ci autofix

**Comment by ilan-gold** [comment]
I can't request a change on a line you didn't change but roughly https://github.com/scverse/scanpy/pull/3299/files#diff-b8aa68d38b59341f74e2277bc6fde44b917d597363ebfd7bf613743d7098a7caR1340-R1344 would become

```python
@pytest.mark.parametrize("color", ["n_genes", "bulk_labels", ["n_genes", "bulk_labels"]])
def test_scatter_no_basis_per_obs(image_comparer, color):
```

**Review by ilan-gold — APPROVED** [review]

**Pull request merged**